example/bookstore/v1/bookstore.proto

```diff
@@ -126,8 +142,16 @@ message Shelf {
 
 // A book resource.
 message Book {
+  option (apigw.v1.message).message_options = {
+    terraform_entity: {
+      name: "Book"
+    }
+  };
   // A unique book id.
-  int64 id = 1;
+  int64 id = 1 [(apigw.v1.field).field_options = {
+    required: true
```

Comment:
hmm. i'm confused about this.

we're saying, this field is "actually" required?

I wonder if we should enforce this instead through protoc-gen-validator annotations?

Reply:
Yeah, this makes it so we can indicate that a given field is required in the openAPI.yaml file. We will also probably want to eventually add some other field options for setting [these](https://speakeasyapi.dev/docs/using-speakeasy/create-terraform/extensions/#x-speakeasy-param-readonly) speakeasy tags on individual params. 

I'm not exactly sure what you are referring to when saying `protoc-gen-validator  annotations`. Where should I look to find this?

Reply:
After discussing this further, I think it’s probably best to follow this pattern for now. It’s not ideal that we have to go back and retroactively add the required tag, but to do it using the validator field from protoc-gen-validate seems like it wouldn’t always work for us. Just for strings we would have to be able to:
1. evaluate if a regex pattern requires at least 1 character
2. check for a min length
3. check for min bytes

Additionally, the bool type and any numeric types do not seem to have a way to indicate that they are required using the validator.

Reply:
Maybe we can call this `terraform-required` or something. Or `required-spec`. IDK, it's just we put it on the api gateway marked as required and don't enforce its existence in the http handler itselt

Reply:
I changed the name and left a comment describing what it is for